File: extra5/app/src/main/java/com/example/extra5/db/NewsDao.java
package com.example.extra5.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface NewsDao {
    @Query("SELECT * FROM news")
    LiveData<List<NewsEntity>> all();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(List<NewsEntity> list);

    @Query("DELETE FROM news")
    void clear();
}

File: extra5/app/src/main/java/com/example/extra5/db/NewsDao.kt
package com.example.extra5.db

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface NewsDao {
    @Query("SELECT * FROM news") fun all(): Flow<List<NewsEntity>>
    @Insert(onConflict = OnConflictStrategy.REPLACE) suspend fun insert(list: List<NewsEntity>)
    @Query("DELETE FROM news") suspend fun clear()
}
